### DISCUSSION TITLE
Add Export Data

### DISCUSSION THREAD

**Discussion by cliver1956** [ainsey11/octopusenergy-consumption-metrics#14] [Q&A]
Hi,
First of all may thanks for your work on this and your blog describing how to implement.
Would it be possible, or is there anything in the development plan, to add electricity export metrics?
Regards
Clive

**Comment by ainsey11** [comment]
Hi Clive!

Apologies for the delayed response, I didn't get a notification from GitHub,

Would you be able to explain this one a little more for me about what it is you'd like to have implemented? The exporter does already provide electricity stats (or should do unless I've broken it!)

Thanks
Rob

**Reply by cliver1956** [reply]
Hi Rob,

No apologies needed. Thank you for your reply and interest in my question.
No need to worry, you haven't broken anything. I am getting data for my gas and electricity use just fine.
The metrics I am seeing in influxdb are as per the attached screenshot...

<img width="341" alt="Octopusdb metrics" src="https://user-images.githubusercontent.com/52658257/217545496-cbc35b9b-2601-4acc-9938-507cadaf7dce.png">

What I am missing is the output from my solar power export as in the attached screenshot.
In essence I have 3 meters in my Octopus dashboard. 1 for gas which is working, 1 for electricity imported from the grid which is working fine and one for electricity exported back to the grid from my solar system. This has the same serial number as my electricity import meter (it's the same physical meter) but a different MPAN.

<img width="562" alt="Octopus Meter Dashboard" src="https://user-images.githubusercontent.com/52658257/217545490-aeb021d4-76a3-498a-ade5-bc9bed9223e4.png">

I'm no expert for sure. But, in essence this would need some additional environment variables along the lines of:
```
OCTO_ELECTRIC_EXPORT_SN = Your electric meter serial number (this would be the same as the import meter ser no.
OCTO_ELECTRIC_MPAN = Your electric export meter MPAN reference
OCTO_ELECTRIC_COST = Your payment per KWH for electricity exported to grid in pence
```

How to implement these additional metrics into your JS code is where I come unstuck.

I hope this makes sense and that it can be implemented. I'm happy to help out in any way I can, testing etc. 
I do have a VERY limited knowledge of JS and am happy, with your guidance, to 'have a bash' at the coding if it helps.
On that score is it true to say that all that's required in your code is to add the additional metrics to the 
```
// Env Vars
const 
```
section 
and the
`console.log(` `
section??

Or am I way off the mark? As I say I'm no expert ;

Anyway, thanks for your time on this.
Regards
Clive

**Reply by ainsey11** [reply]
Hi Clive,

That now makes a lot more sense, thanks for taking the time to explain it to me,
I've not got solar at my house so I wasn't aware that the API could provide the metrics for it,

I'd be more than happy to implement the feature, I just need some way of either getting an example response from the octopus API or being able to interact with it for an account that has the data available as my account is just a normal meter,

I think what I may do, is make a discord server for the project so people can discuss things like this / share data without it being publicly viewable (I don't want your SN to be leaked for example) - do you use discord?

**Reply by cliver1956** [reply]
Hi Rob, I don't currently use discord but there is no reason why I can't.
Interestingly I have just had a look at my API access page on Octopus at https://octopus.energy/dashboard/developer/
to see if I could download the data you need.
Sadly it only show the curl commands for gas and electricity consumption. I assume it is a case of changing the MPAN and /consumption element to /export or something similar.
I have emailed Octopus support for clarification.
Once I have a reply from them I'll come back here. Happy to give you access to my API via a secure means when you need it.

**Reply by cliver1956** [reply]
I just tried changing the MPAN and /consumption for /export on the off chance but it didn't work :(

**Reply by ainsey11** [reply]
Thanks Clive - I think it would be helpful if you wouldn't mind sharing your key with me so I can see what sort of responses the API gives us to work with, with any luck support will give us some useful information too,

I've just spun up a discord server and popped the link in the readme - if you catch 5 mins, feel free to join and we can have a chat and see what we can figure out :) 

Cheers
Rob
